CONTRIBUTING.md

```diff
@@ -198,50 +198,54 @@ section of the Qiskit documentation.
 
 ### Test
 
-New features often imply changes in the existent tests or new ones are
-needed. Once they're updated/added run this be sure they keep passing.
+#### Test Types
+There are three different types of tests in `qiskit-ibm-runtime`. The implementation is based upon the well-documented [unittest](https://docs.python.org/3/library/unittest.html) Unit testing framework.
 
-For executing the tests, a `make test` target is available.
+##### 1. Unit tests
+Run locally without connecting to an external system. They are short-running, stable and give a basic level of confidence during development.
 
-For executing a simple python test manually, you can just run this
-command:
-
-Linux and Mac:
+To execute all unit tests, run:
+``` {.bash}
+$ python -m unittest discover --verbose --top-level-directory . --start-directory test/types/unit
+```
+##### 2. Integration tests
+Executed against an external system configured via a (token, instance, url) tuple. Detailed coverage of happy and non-happy paths. They are long-running and unstable at times. A successful test run gives a high level of confidence that client and APIs work well together.
 
+To execute all integration tests, run
 ``` {.bash}
-$ python -m unittest test/test_something.py
+$ python -m unittest discover --verbose --top-level-directory . --start-directory test/types/integration
 ```
 
-Windows:
+##### 3. E2E tests
+
+Executed against an external system configured via a (token, instance, url) tuple. Basic coverage of most important user-facing happy paths. Test suite runs faster than integration but slower than unit tests and is stable.
```

Comment:
```suggestion
Executed against an external system configured via a (token, instance, url) tuple. Basic coverage of most important user-facing happy and non-happy paths. Test suite runs faster than integration but slower than unit tests and is stable.
```

Reply:
My understanding (and the current status quo) is different here.

We have extensive test coverage for integration tests (with happy, non-happy paths). We therefore accept them to be instable at times.

For our e2e tests the goal should be to keep them simple and focus on the most common user-facing paths without covering non-happy paths.

I'm fine with updating the definition if we reach another agreemetn.